fastdds_python/test/api/test_qos.py

```diff
@@ -0,0 +1,1409 @@
+        elif 2 == count:
+            assert(1 == user_value)
+        elif 3 == count:
+            assert(2 == user_value)
+        else:
+            assert(3 == user_value)
+        count += 1
+
+    # .ownership
+    datareader_qos.ownership().kind = fastdds.EXCLUSIVE_OWNERSHIP_QOS
+    assert(fastdds.EXCLUSIVE_OWNERSHIP_QOS == datareader_qos.ownership().kind)
+
+    # .time_based_filter
+    datareader_qos.time_based_filter().minimum_separation.seconds = fastdds.Time_t.INFINITE_SECONDS
+    datareader_qos.time_based_filter().minimum_separation.nanosec = fastdds.Time_t.INFINITE_NANOSECONDS
+    assert(fastdds.Time_t.INFINITE_SECONDS == datareader_qos.time_based_filter().minimum_separation.seconds)
+    assert(fastdds.Time_t.INFINITE_NANOSECONDS == datareader_qos.time_based_filter().minimum_separation.nanosec)
+
+    # .reader_data_lifecycle
+    datareader_qos.reader_data_lifecycle().autopurge_disposed_samples_delay.seconds = 100
+    datareader_qos.reader_data_lifecycle().autopurge_disposed_samples_delay.nanosec = 30000
+    datareader_qos.reader_data_lifecycle().autopurge_no_writer_samples_delay.seconds = 30000
+    datareader_qos.reader_data_lifecycle().autopurge_no_writer_samples_delay.nanosec = 100
+    assert(100 == datareader_qos.reader_data_lifecycle().autopurge_disposed_samples_delay.seconds)
+    assert(30000 == datareader_qos.reader_data_lifecycle().autopurge_disposed_samples_delay.nanosec)
+    assert(30000 == datareader_qos.reader_data_lifecycle().autopurge_no_writer_samples_delay.seconds)
+    assert(100 == datareader_qos.reader_data_lifecycle().autopurge_no_writer_samples_delay.nanosec)
+
+    # .lifespan
+    datareader_qos.lifespan().duration.seconds = 10
+    datareader_qos.lifespan().duration.nanosec = 33
+    assert(10 == datareader_qos.lifespan().duration.seconds)
+    assert(33 == datareader_qos.lifespan().duration.nanosec)
+
+    # .durability_service
+    datareader_qos.durability_service().history_kind = fastdds.KEEP_ALL_HISTORY_QOS
+    datareader_qos.durability_service().history_depth = 10
+    datareader_qos.durability_service().max_samples = 5
+    datareader_qos.durability_service().max_instances = 20
+    datareader_qos.durability_service().max_samples_per_instance = 30
+    assert(fastdds.KEEP_ALL_HISTORY_QOS == datareader_qos.durability_service().history_kind)
+    assert(10 == datareader_qos.durability_service().history_depth)
+    assert(5 == datareader_qos.durability_service().max_samples)
+    assert(20 == datareader_qos.durability_service().max_instances)
+    assert(30 == datareader_qos.durability_service().max_samples_per_instance)
+
+    # .reliable_reader_qos
+    datareader_qos.reliable_reader_qos().times.initialAcknackDelay.seconds = 34
+    datareader_qos.reliable_reader_qos().times.initialAcknackDelay.nanosec = 32
+    datareader_qos.reliable_reader_qos().times.heartbeatResponseDelay.seconds = 432
+    datareader_qos.reliable_reader_qos().times.heartbeatResponseDelay.nanosec = 43
```

Comment:
I don't see any mention of the `times` member in the Python API.

Reply:
I will check it

Reply:
`ReaderTimes` is from *RTPS* API (see [this](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/rtps/attributes/ReaderTimes.html?highlight=ReaderTimes)). Python bindings allow to use the *DDS* API and therefore this API is the documented one. But this API uses some times *RTPS* structures. We can talk about this in a meeting.